src/analytics/mod.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::types::{
+    ledger::BlockMetadata,
+    stardust::block::{Block, Output},
+    tangle::MilestoneIndex,
+};
+
+pub mod address_balance;
+pub mod block_activity;
+
+#[allow(missing_docs)]
+pub trait BlockAnalytics {
+    type Measurement;
+    fn begin_milestone(&mut self, index: MilestoneIndex);
+    fn handle_block(&mut self, block: &Block, block_metadata: &BlockMetadata, inputs: &Option<Vec<Output>>);
```

Comment:
shouldn't `inputs` rather be `Option<&[Output]>`?

Reply:
We might not even need the `inputs` here at all. Everything related to the ledger should probably be handled by `TransactionAnalytics`.

Reply:
Makes sense.